hibernate-core/src/test/java/org/hibernate/test/flush/TestHibernateFlushModeOnInactiveTransaction.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * Hibernate, Relational Persistence for Idiomatic Java
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later.
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.test.flush;
+
+import org.hibernate.FlushMode;
+import org.hibernate.Session;
+import org.hibernate.testing.TestForIssue;
+import org.hibernate.testing.junit4.BaseCoreFunctionalTestCase;
+import org.junit.Test;
+
+/**
+ * Test for issue https://hibernate.atlassian.net/browse/HHH-13663
+ * 
+ * @author Luca Domenichini
+ */
+@TestForIssue(jiraKey = "HHH-13663")
+public class TestHibernateFlushModeOnInactiveTransaction extends BaseCoreFunctionalTestCase {
+
+	@Test
```

Comment:
So the problem is actually only triggered when using thread-local bound Sessions.

I think the test should enable that as it's not the default?